json/src/main/java/io/airlift/json/RecordAutoDetectModule.java

```diff
@@ -45,4 +53,134 @@ public Version version()
             return Version.unknownVersion();
         }
     }
+
+    private static class RecordVisibilityChecker
+            implements VisibilityChecker<RecordVisibilityChecker>
+    {
+        private final Set<String> componentNames;
+
+        RecordVisibilityChecker(Class<? extends Record> recordClass)
+        {
+            componentNames = Stream.of(recordClass.getRecordComponents())
+                    .map(RecordComponent::getName)
+                    .collect(toImmutableSet());
+        }
+
+        @Override
+        public RecordVisibilityChecker with(JsonAutoDetect ann)
```

Comment:
Avoid abbreviations

Reply:
good catch! this matches parameter names in jackson interface we're implementing here, but i will change them nonetheless.